tasklist/webapp/src/main/java/io/camunda/tasklist/webapp/api/rest/v1/controllers/TaskController.java

```diff
@@ -230,7 +230,25 @@ private void checkTaskImplementation(final LazySupplier<TaskDTO> taskSupplier) {
   }
 
   private boolean hasAccessToTask(final LazySupplier<TaskDTO> taskSupplier) {
-    return true;
+    final String userName = userReader.getCurrentUserId();
+    final List<String> listOfUserGroups = identityAuthorizationService.getUserGroups();
+    final var task = taskSupplier.get();
+    final boolean allUsersTask =
+        task.getCandidateUsers() == null && task.getCandidateGroups() == null;
+    final boolean candidateGroupTasks =
+        task.getCandidateGroups() != null
+            && !Collections.disjoint(Arrays.asList(task.getCandidateGroups()), listOfUserGroups);
+    final boolean candidateUserTasks =
+        task.getCandidateUsers() != null
+            && Arrays.asList(task.getCandidateUsers()).contains(userName);
+    final boolean assigneeTasks = task.getAssignee() != null && task.getAssignee().equals(userName);
+    final boolean noRestrictions = listOfUserGroups.contains(IdentityProperties.FULL_GROUP_ACCESS);
```

Comment:
🔧  this is actually not supported with the new group membership setup 🤔  could we remove it already alongside relevant tests?

Reply:
Lets do a quick sync on this one. 
From my understanding V1 should still support Task Restrictions, but lets have a quick chat to align this.

Reply:
to calrify, this was about the special value of `FULL_GROUP_ACCESS` granting access to all groups

Reply:
The `FULL_GROUP_ACCESS`  will return an empty string, and it is an special value, used in case of the fallback.
This wont cause issues for consolidated-auth, once it is not a valid value over there. 

When the `DefaultUserGroupService` is no longer necessary, this can be removed, once we can rely only in the `CamundaUserService` Groups